commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitTest.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.commons.test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.testng.ITestResult;
+
+public class PolarionJUnitTest {
+
+   public enum Status {SUCCESS, FLAKY, FAILURE, ERROR, SKIPPED}
+
+   final String name;
+   final String clazz;
+   final List<Throwable> failures = new ArrayList<>();
+   final AtomicInteger successes =  new AtomicInteger();
+   final AtomicLong elapsedTime = new AtomicLong();
+
+   volatile Status status = Status.ERROR;
+
+   public PolarionJUnitTest(String name, String clazz) {
+      this(name, clazz, null);
+   }
+
+   public PolarionJUnitTest(String name, String clazz, Throwable t) {
+      this.name = name;
+      this.clazz = clazz;
+      if (t != null) {
+         this.failures.add(t);
+         this.status = Status.FAILURE;
+      }
+   }
+
+   void add(ITestResult tr) {
+      switch (tr.getStatus()) {
+         case ITestResult.SUCCESS:
+            status = failures.isEmpty() ? Status.SUCCESS : Status.FLAKY;
+            successes.incrementAndGet();
+            break;
+         case ITestResult.FAILURE:
+            failures.add(tr.getThrowable());
+            status = Status.FAILURE;
```

Comment:
shouldn't be `FLAKY` if `successes > 0`?

Reply:
Flaky tests are executed until they pass, or the max number of attempts has been reached. Therefore for a test to be FLAKY, the last encountered test must be a success.